tasks/ssh.yml

```diff
@@ -15,3 +15,15 @@
     key: "{{ keydata.stdout }}"
     state: present
   delegate_to: "{{ kdump_ssh_server }}"
+
+- name: fetch the servers public key
+  slurp:
+    src: /etc/ssh/ssh_host_rsa_key.pub
+  register: serverpubkey
+  delegate_to: "{{ kdump_ssh_server }}"
```

Comment:
This will most likely not work in check mode. Cf. https://github.com/linux-system-roles/kdump/pull/23#discussion_r279333359

Reply:
Or maybe it will, just like stat. https://github.com/linux-system-roles/kdump/pull/23/files/4ca107cd39559e1beffa3e2e096995b9fd69755a#r279344478

Reply:
So we're grabbing the ssh pub key from the managed host and adding that to the known_hosts file on the managed host?  In order to allow the managed host to ssh to itself?

Reply:
there are two managed hosts: the one that will kdump and the other that will store the coredump (if transmitting the coredump over ssh). The former is a ssh client, the other a server.

Reply:
The test is a degenerate case and uses a single host for the two, but in practice this will not be the case.

Reply:
> The test is a degenerate case and uses a single host for the two, but in practice this will not be the case.

ok - then should this code be in the test code instead of the main line code?

Reply:
no, the intent is that this be the functionality be in the role, so that users benefit from it.

When this functionality is missing, kdump does not start, because it can not contact the ssh server: ssh is asking to confirm the passphrase.

Reply:
the code here works properly both in the test case where server and client are the same host, and in the real case where they are not.

Reply:
> there are two managed hosts: the one that will kdump and the other that will store the coredump

Not entirely accurate: the one that will store the coredump should not be targeted by the role (the playbook using the role), the only place where Ansible will manage it is this delegated task. Only the one(s) that will kdump should be targeted by the playbook.